core/src/main/scala/org/apache/spark/util/DirectByteBufferOutputStream.scala

```diff
@@ -63,15 +65,29 @@ private[spark] class DirectByteBufferOutputStream(capacity: Int) extends OutputS
     buffer = newBuffer
   }
 
-  def reset(): Unit = buffer.clear()
+  private def checkNotClosed(): Unit = {
+    if (buffer == null) {
+      throw new IllegalStateException(
```

Comment:
Could you replace this by `SparkException.internalError`, please.

Reply:
Done.